imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+
+DEFAULT_GLOBAL_CDF_ATTRS_FILE = "imap_default_global_cdf_attrs.yaml"
+DEFAULT_GLOBAL_CDF_ATTRS_SCHEMA_FILE = "hermes_default_global_cdf_attrs_schema.yaml"
+DEFAULT_VARIABLE_CDF_ATTRS_SCHEMA_FILE = "hermes_default_variable_cdf_attrs_schema.yaml"
+
+class CdfAttributeManager:
+    def __init__(self, data_dir: Path):
+
+        # TODO: Split up schema source and data source?
+        self.source_dir = data_dir
+
+        # TODO: copied from hermes_core. Currently we can use default schema, but
+        # We should add some way of extending the schema and remove all the HERMES specific stuff
+        # Data Validation, Complaiance, Derived Attributes
+        self.global_attribute_schema = self._load_default_global_attr_schema()
+
+        # Data Validation and Compliance for Variable Data
+        self.variable_attribute_schema = (
+            self._load_default_variable_attr_schema()
+        )
+
+        self.cdftypenames = {
+            const.CDF_BYTE.value: "CDF_BYTE",
+            const.CDF_CHAR.value: "CDF_CHAR",
+            const.CDF_INT1.value: "CDF_INT1",
+            const.CDF_UCHAR.value: "CDF_UCHAR",
+            const.CDF_UINT1.value: "CDF_UINT1",
+            const.CDF_INT2.value: "CDF_INT2",
+            const.CDF_UINT2.value: "CDF_UINT2",
+            const.CDF_INT4.value: "CDF_INT4",
+            const.CDF_UINT4.value: "CDF_UINT4",
+            const.CDF_INT8.value: "CDF_INT8",
+            const.CDF_FLOAT.value: "CDF_FLOAT",
+            const.CDF_REAL4.value: "CDF_REAL4",
+            const.CDF_DOUBLE.value: "CDF_DOUBLE",
+            const.CDF_REAL8.value: "CDF_REAL8",
+            const.CDF_EPOCH.value: "CDF_EPOCH",
+            const.CDF_EPOCH16.value: "CDF_EPOCH16",
+            const.CDF_TIME_TT2000.value: "CDF_TIME_TT2000",
+        }
+        self.numpytypedict = {
+            const.CDF_BYTE.value: np.int8,
+            const.CDF_CHAR.value: np.int8,
+            const.CDF_INT1.value: np.int8,
+            const.CDF_UCHAR.value: np.uint8,
+            const.CDF_UINT1.value: np.uint8,
+            const.CDF_INT2.value: np.int16,
+            const.CDF_UINT2.value: np.uint16,
+            const.CDF_INT4.value: np.int32,
+            const.CDF_UINT4.value: np.uint32,
+            const.CDF_INT8.value: np.int64,
+            const.CDF_FLOAT.value: np.float32,
+            const.CDF_REAL4.value: np.float32,
+            const.CDF_DOUBLE.value: np.float64,
+            const.CDF_REAL8.value: np.float64,
+            const.CDF_EPOCH.value: np.float64,
+            const.CDF_EPOCH16.value: np.dtype((np.float64, 2)),
+            const.CDF_TIME_TT2000.value: np.int64,
+        }
+        self.timetypes = [
+            const.CDF_EPOCH.value,
+            const.CDF_EPOCH16.value,
+            const.CDF_TIME_TT2000.value,
+        ]
+
+        # Load Default IMAP Global Attributes
+        self.global_attributes = self._load_yaml_data(
+            str(self.source_dir / DEFAULT_GLOBAL_CDF_ATTRS_FILE)
```

Comment:
You are casting to a lot of strings. If you already have a Path it looks like you should be able to keep it as a Path in your loader function as well already I think. I'd suggest removing all str() casts throughout.
```suggestion
            self.source_dir / DEFAULT_GLOBAL_CDF_ATTRS_FILE
```

Reply:
the move to accept Path types was a late addition, but agree with this so I will update it.